server/application-server/src/main/java/de/tum/in/www1/hephaestus/codereview/user/UserRepository.java

```diff
@@ -0,0 +1,29 @@
+package de.tum.in.www1.hephaestus.codereview.user;
+
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+
+    @Query("SELECT u FROM User u WHERE u.login = :login")
+    Optional<User> findUser(@Param("login") String login);
+
+    @Query("""
+                SELECT new de.tum.in.www1.hephaestus.codereview.user.UserDTO(u.login, u.email, u.name, u.url)
```

Comment:
Apparently we can use the `ClassImportIntegrator` utility from [Hypersistence Utils project](https://github.com/vladmihalcea/hypersistence-utils) to avoid having to write fully-qualified names of Java classes.

```
public class ClassImportIntegratorIntegratorProvider implements IntegratorProvider {
 
    @Override
    public List<Integrator> getIntegrators() {
        return List.of(
            new ClassImportIntegrator(
                List.of(
                    PostDTO.class
                )
            )
        );
    }
}
```

And then in `application.yaml`:
```
spring:
  jpa:
    properties:
      hibernate:
        integrator_provider=your.awesome.app.ClassImportIntegratorIntegratorProvider
```

Can you try that?
Here is the blog post where I found it:
https://vladmihalcea.com/dto-projection-jpa-query/